### PR TITLE
Welcome Flow: Create Account Designs

### PR DESCRIPTION
**Pull request by Kokobe**
Implements the Create Account Screens. 

* Forks the buttons from FirstRun and repurposes them.
* Functionality happening in the next PR.

### Pointers to help with review
* https://www.figma.com/file/dylvDPB69JG9U1rzZ3HQ52/Android-Premium?node-id=7%3A1207&t=3IcJbiz1fpC8ytsS-0

### How was this tested?
- [x] I tested this manually on `debug`
- [ ] I tested this manually on `staging` or `release`
- [ ] I added automated tests
- [ ] Existing automated tests cover some of my changes
- [ ] Existing automated tests cover all of my changes
- [x] I checked that `@Previews` still work in Android Studio

### Screenshots and screen recordings

Screen Shot 2022-12-17 at 2 02 32 AM
Screen Shot 2022-12-17 at 2 02 37 AM
Screen Shot 2022-12-17 at 2 35 28 AM
Screen Shot 2022-12-17 at 2 35 33 AM

### CODE REVIEW

File: app/src/main/java/com/neeva/app/welcomeflow/SharedComposables.kt [hunk 1]
```diff
@@ -61,27 +62,39 @@ fun WelcomeFlowStackedButtons(
...
+    WelcomeFlowButtonContainer {
+        WelcomeFlowButton(primaryText = primaryText, onClick = onPrimaryButton)
+        Spacer(Modifier.height(18.dp))
+        SecondaryWelcomeFlowButton(text = secondaryText, onClick = onSecondaryButton)
+    }
+}
+
+@Composable
+fun SecondaryWelcomeFlowButton(text: String, onClick: () -> Unit) {
+    Button(
```
Comment (dan-alcantara-neeva) [inline app/src/main/java/com/neeva/app/welcomeflow/SharedComposables.kt:74]: Isn't there `TextButton` that has the style that you want (just blue on white)?  Then you don't have to hard-code these colors.

File: app/src/main/java/com/neeva/app/welcomeflow/WelcomeFlowContainer.kt [hunk 2]
```diff
@@ -113,7 +114,7 @@ fun WelcomeFlowHeader(
                 Spacer(Modifier.height(Dimensions.PADDING_LARGE))
                 WordMark(
                     colorFilter = ColorFilter.tint(LocalContentColor.current),
-                    modifier = Modifier.height(20.dp)
+                    modifier = Modifier.width(89.dp)
```
Comment (dan-alcantara-neeva) [inline app/src/main/java/com/neeva/app/welcomeflow/WelcomeFlowContainer.kt:117]: 80 I can deal with, but this 89 is super specific.  Confirm that it's an intentional value with @hwiatneeva and that the number in the Figma isn't calculated by a rule being applied (like centered within the container).

File: app/src/main/java/com/neeva/app/welcomeflow/createaccount/LoginButtons.kt [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    if (provider == NeevaUser.SSOProvider.GOOGLE) {
+        LoginButton(
+            text = getSSOProviderOnboardingText(provider, true),
+            startComposable = { SSOProviderButtonIcon(ssoProvider = provider) },
+            containerColor = MaterialTheme.colorScheme.primary,
+            contentColor = MaterialTheme.colorScheme.onPrimary,
+            onClick = { onLaunchLoginFlow(resultLauncher, launchLoginFlowParams) }
+        )
+    } else {
+        LoginButton(
```
Comment (dan-alcantara-neeva) [inline app/src/main/java/com/neeva/app/welcomeflow/createaccount/LoginButtons.kt:81]: This is basically an `OutlinedButton`, isn't it?  https://m3.material.io/components/buttons/guidelines

File: app/src/main/java/com/neeva/app/welcomeflow/createaccount/LoginButtons.kt [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+            contentColor = MaterialTheme.colorScheme.primary,
+            border = BorderStroke(1.dp, MaterialTheme.colorScheme.outline),
+            onClick = { onLaunchLoginFlow(resultLauncher, launchLoginFlowParams) }
+        )
+    }
+}
+
+@Composable
+fun getSSOProviderOnboardingText(provider: NeevaUser.SSOProvider, signup: Boolean): String {
+    when (provider) {
```
Comment (dan-alcantara-neeva) [inline app/src/main/java/com/neeva/app/welcomeflow/createaccount/LoginButtons.kt:94]: Maybe pull the `return stringResource` up to the `when`?  Then you don't have to keep repeating it.

```
return stringResource(
    when (provider) {
        NeevaUser.SSOProvider.MICROSOFT -> {
            if (signup) {
                R.string.sign_up_with_microsoft
            } else {
                R.string.sign_in_with_microsoft
            }
        }

        NeevaUser.SSOProvider.GOOGLE -> {
          ...
    }
}
```

File: app/src/main/java/com/neeva/app/welcomeflow/createaccount/SSOProviderButtonIcon.kt [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.res.painterResource
+import com.neeva.app.R
+import com.neeva.app.ui.theme.Dimensions
+import com.neeva.app.userdata.NeevaUser
+
+@Composable
+fun SSOProviderButtonIcon(ssoProvider: NeevaUser.SSOProvider, useIcon: Boolean = false) {
+    when (ssoProvider) {
```
Comment (dan-alcantara-neeva) [inline app/src/main/java/com/neeva/app/welcomeflow/createaccount/SSOProviderButtonIcon.kt:20]: Is there any difference between these cases?  Might as well combine all three provider types.

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -515,6 +515,7 @@
     <string name="sign_up_with_google">Sign up with Google</string>
     <string name="sign_up_with_microsoft">Sign up with Microsoft</string>
     <string name="sign_up_other_options">Other sign-up options</string>
+    <string name="sign_up_with_email">Sign up with Email</string>
```
Comment (dan-alcantara-neeva) [inline app/src/main/res/values/strings.xml:518]: Make sure that `Email` is supposed to be capitalized.  We lowercase it on our privacy policy and login screens (Please enter your email address), the error says "You already have an account with this email address", etc.

**Review by dan-alcantara-neeva — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/neeva/app/welcomeflow/WelcomeFlowContainer.kt (hunk 2 above)
Comment (Kokobe) [reply app/src/main/java/com/neeva/app/welcomeflow/WelcomeFlowContainer.kt:117]: Screen Shot 2022-12-19 at 11 25 48 AM

Hwi is out this week, but the designs (https://www.figma.com/file/dylvDPB69JG9U1rzZ3HQ52/Android-Premium?node-id=524%3A3564&t=I4tngfbUKr8DsB7M-0) say 89.17 😅 

There aren't any rules or special container centering/sizing

File: app/src/main/java/com/neeva/app/welcomeflow/createaccount/LoginButtons.kt (hunk 3 above)
Comment (Kokobe) [reply app/src/main/java/com/neeva/app/welcomeflow/createaccount/LoginButtons.kt:81]: Yes, but I'm trying to reuse the same code for the Google sign-in button, which just uses a different button coloring style:

Screen Shot 2022-12-19 at 11 31 40 AM

File: app/src/main/java/com/neeva/app/welcomeflow/createaccount/SSOProviderButtonIcon.kt (hunk 5 above)
Comment (Kokobe) [reply app/src/main/java/com/neeva/app/welcomeflow/createaccount/SSOProviderButtonIcon.kt:20]: Ah one uses an `Image` and the other uses an `Icon`

File: app/src/main/res/values/strings.xml (hunk 6 above)
Comment (Kokobe) [reply app/src/main/res/values/strings.xml:518]: Screen Shot 2022-12-19 at 11 37 30 AM

Comment (dan-alcantara-neeva) [reply app/src/main/res/values/strings.xml:518]: Yeah, I know what the mocks say.  Is this a mistake in the mocks?

Comment (dan-alcantara-neeva) [reply app/src/main/res/values/strings.xml:518]: Lowercase the e, based on iOS.  Don't want to have to deal with updating Smartling translations if we can correct the mistake now.

**Review by dan-alcantara-neeva — APPROVED** [review]

File: app/src/main/java/com/neeva/app/welcomeflow/WelcomeFlowContainer.kt (hunk 2 above)
Comment (hwiatneeva) [reply app/src/main/java/com/neeva/app/welcomeflow/WelcomeFlowContainer.kt:117]: Sorry here 100x20

neeva-wordmark-100x20

File: app/src/main/res/values/strings.xml (hunk 6 above)
Comment (hwiatneeva) [reply app/src/main/res/values/strings.xml]: Yes sorry -- 'e'!

**Pull request merged**